### PR TITLE
adjust the expected output based kubectl verison

### LINKED ISSUE #56414: [e2e failure] [sig-cli] Kubectl client [k8s.io] Kubectl describe should check if kubectl describe prints relevant information for rc and pods [Conformance]
/priority critical-urgent
/priority failing-test
/kind bug
/status approved-for-milestone
@kubernetes/sig-cli-test-failures 

This test has been failing since at least 2017-11-23 in the following jobs:
- https://k8s-testgrid.appspot.com/sig-release-master-upgrade#gce-1.8-master-upgrade-master
- https://k8s-testgrid.appspot.com/sig-release-master-upgrade#gce-1.8-master-upgrade-cluster-parallel
- https://k8s-testgrid.appspot.com/sig-release-master-upgrade#gce-master-1.8-downgrade-cluster-parallel
- https://k8s-testgrid.appspot.com/sig-release-master-upgrade#gke-gci-master-gci-1.8-downgrade-cluster-parallel

These jobs are on the [sig-release-master-upgrade dashboard](https://k8s-testgrid.appspot.com/sig-release-master-upgrade), and prevent us from cutting [v1.9.0-beta.1] (kubernetes/sig-release#34). Is there work ongoing to bring this test back to green?

triage cluster [9171dbbaaa771d52dba5](https://storage.googleapis.com/k8s-gubernator/triage/index.html?test=%5C%5Bsig%5C-cli%5C%5D%5C%20Kubectl%5C%20client%5C%20%5C%5Bk8s%5C.io%5C%5D%5C%20Kubectl%5C%20describe%5C%20should%5C%20check%5C%20if%5C%20kubectl%5C%20describe%5C%20prints%5C%20relevant%5C%20information%5C%20for%5C%20rc%5C%20and%5C%20pods%5C%20%5C%5BConformance%5C%5D#9171dbbaaa771d52dba5)
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl/kubectl.go:934
Nov 21 15:24:34.066: failed to find Created By: in Name:           redis-master-fjtpl
Namespace:      e2e-tests-kubectl-h9s9j
```

- also failing in:
  - https://k8s-testgrid.appspot.com/sig-release-master-kubectl-skew#skew-clustermaster-kubectl1.8-gce
  - https://k8s-testgrid.appspot.com/sig-release-master-kubectl-skew#skew-cluster1.8-kubectl1.8-gke
- Related issues or PRs:
  - https://github.com/kubernetes/kubernetes/issues/43464
  - https://github.com/kubernetes/kubernetes/issues/41898

### PR DESCRIPTION
**Pull request by mengqiy**
Fixes #56414

```release-note
NONE
```

/assign @spiffxp

### CODE REVIEW

**Comment by mengqiy** [comment]
@spiffxp @pwittrock PTAL

**Comment by mengqiy** [comment]
/assign @janetkuo 
because of https://github.com/kubernetes/kubernetes/pull/42849

File: test/e2e/kubectl/kubectl.go [hunk 1]
```diff
@@ -857,12 +860,16 @@ metadata:
...
+				if !gte {
+					requiredStrings = append(requiredStrings, []string{"Created By:", "ReplicationController/redis-master"})
+				}
+				requiredStrings = append(requiredStrings, [][]string{
 					{"Controlled By:", "ReplicationController/redis-master"},
 					{"Image:", redisImage},
 					{"State:", "Running"},
 					{"QoS Class:", "BestEffort"},
-				}
+				}...)
```
Comment (janetkuo) [inline test/e2e/kubectl/kubectl.go]: From `checkOutput` code, I don't think the order matters. Would you check again? If so, we can just append it at the end.

File: test/e2e/kubectl/kubectl.go [hunk 2]
```diff
@@ -847,6 +847,9 @@ metadata:
 
 			// Pod
 			forEachPod(func(pod v1.Pod) {
+				gte, err := framework.KubectlVersionGTE(utilversion.MustParseSemantic("v1.9.0-alpha.0"))
```
Comment (janetkuo) [inline test/e2e/kubectl/kubectl.go]: This is affected by https://github.com/kubernetes/kubernetes/commit/efbfead4ef9fd91ce130586787a6e3c54ca21e02#diff-fde2ce0a73639d19347a5ea6fe001ba6 which is in `v1.9.0-alpha.2`

**Comment by janetkuo** [comment]
cc @crimsonfaith91: kubectl skew test failed bue to the removal of created-by annotation

File: test/e2e/kubectl/kubectl.go (hunk 2 above)
Comment (mengqiy) [reply test/e2e/kubectl/kubectl.go]: Done

File: test/e2e/kubectl/kubectl.go (hunk 1 above)
Comment (mengqiy) [reply test/e2e/kubectl/kubectl.go]: The order matters here.
The index of the actual output always increases after a match, so the order matters.
https://github.com/mengqiy/kubernetes/blob/3d5ff42cc5cc0a71668e9490ae8cb14b644d02e7/test/e2e/kubectl/kubectl.go#L1759

File: test/e2e/kubectl/kubectl.go (hunk 2 above)
Comment (janetkuo) [reply test/e2e/kubectl/kubectl.go]: Server version should be checked instead

File: test/e2e/kubectl/kubectl.go [hunk 3]
```diff
@@ -847,6 +847,9 @@ metadata:
 
 			// Pod
 			forEachPod(func(pod v1.Pod) {
+				gte, err := framework.KubectlVersionGTE(utilversion.MustParseSemantic("v1.9.0-alpha.2"))
```
Comment (janetkuo) [inline test/e2e/kubectl/kubectl.go]: Server version should be checked instead

File: test/e2e/kubectl/kubectl.go (hunk 1 above)
Comment (janetkuo) [reply test/e2e/kubectl/kubectl.go]: ok

File: test/e2e/kubectl/kubectl.go (hunk 3 above)
Comment (mengqiy) [reply test/e2e/kubectl/kubectl.go]: Why checking server version? The describe behavior is purely client-side, isn't?

**Comment by janetkuo** [comment]
In failing tests, 1.8 clients are tested against 1.9 servers (in fact 1.10/master, but technically it's 1.9), see log here: https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-master-new-gci-kubectl-skew/3839/build-log.txt

As I mentioned in https://github.com/kubernetes/kubernetes/issues/56414#issuecomment-348006382, this is affected by #54445 (1.9), where `created-by` annotation generation was removed from server side: https://github.com/kubernetes/kubernetes/pull/54445/files#diff-44bba2c68a6ea2b4ca0dbb7f4465828eL498

The failure is because 1.9 server doesn't add those created-by annotations, therefore when 1.8 client parse sannotations, it won't print "Created By: xxx".

File: test/e2e/kubectl/kubectl.go (hunk 3 above)
Comment (janetkuo) [reply test/e2e/kubectl/kubectl.go]: Responded here: https://github.com/kubernetes/kubernetes/pull/56508#issuecomment-348029287

Comment (janetkuo) [reply test/e2e/kubectl/kubectl.go]: Actually, skip both server and client.

**Comment by mengqiy** [comment]
> Actually, check both server and client. Neither 1.9 server nor 1.9 client will print "Created By:" in kubectl describe against any server/clients.

Done.

File: test/e2e/kubectl/kubectl.go [hunk 4]
```diff
@@ -847,6 +847,10 @@ metadata:
 
 			// Pod
 			forEachPod(func(pod v1.Pod) {
+				servergte, err := framework.ServerVersionGTE(utilversion.MustParseSemantic("v1.9.0-alpha.2"), c.Discovery())
```
Comment (janetkuo) [inline test/e2e/kubectl/kubectl.go]: this err needs to be checked too

**Comment by janetkuo** [comment]
One last comment and LGTM otherwise

File: test/e2e/kubectl/kubectl.go (hunk 4 above)
Comment (mengqiy) [reply test/e2e/kubectl/kubectl.go]: Done

**Comment by crimsonfaith91** [comment]
Everything looks good to me!
Thanks @mengqiy for working on this!

**Comment by mengqiy** [comment]
Someone please apply the lgtm label.

**Comment by mengqiy** [comment]
@jpbetz Can you please put this in the 1.8 milestone?

**Comment by mengqiy** [comment]
The bot is asking for so many labels :-/

> This pull request must have the status/approved-for-milestone label applied by a SIG maintainer.

The bot is asking for `status/approved-for-milestone` label. @janetkuo Can you help me with that since you are one of the sig maintainers.

**Pull request merged**